installers/msi-language/ActiveState/Tracker.cs

```diff
@@ -81,15 +82,46 @@ public async Task TrackS3Event(Session session, string sessionID, string categor
             session.Log("Successfully downloaded S3 pixel string");
         }
 
-        private string computeSessionID(string msiLogFileName)
+        internal class GACustomDimensions
         {
-            using (var md5 = System.Security.Cryptography.MD5.Create())
+            public string productVersion;
+            public string sessionID;
+            public string uiLevel;
+            public string installMode;
+
+            public GACustomDimensions(Session session, string cid)
+            {
+                this.productVersion = getValueFromSession(session, "ProductVersion", "PRODUCT_VERSION");
+                var msiLogFileName = getValueFromSession(session, "MsiLogFileLocation", "MsiLogFileLocation");
+                this.sessionID = computeSessionID(cid, msiLogFileName);
+                this.uiLevel = getValueFromSession(session, "UILevel", "UI_LEVEL");
+                this.installMode = getValueFromSession(session, "INSTALL_MODE", "INSTALL_MODE");
+            }
+
+            private string computeSessionID(string cid, string msiLogFileName)
             {
-                byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(this._cid + msiLogFileName);
-                byte[] hashBytes = md5.ComputeHash(inputBytes);
+                using (var md5 = System.Security.Cryptography.MD5.Create())
+                {
+                    byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(cid + msiLogFileName);
+                    byte[] hashBytes = md5.ComputeHash(inputBytes);
+
+                    return new Guid(hashBytes).ToString();
+                }
+            }
 
-                return new Guid(hashBytes).ToString();
+
+            private string getValueFromSession(Session session, string key1, string key2)
+            {
+                if (!session.GetMode(InstallRunMode.Scheduled))
+                {
+                    return session[key1];
+                }
+                if (session.CustomActionData.ContainsKey(key2)) {
+                    return session.CustomActionData[key2];
+                }
+                return "";
```

Comment:
Might be worthwhile surrounding this in a try-catch to ensure we don't get an exception for a key error, even though that is very unlikely.

On that note I don't think we're sending the `INSTALL_MODE` property on uninstall. Is that a value we want to capture?